src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -1,206 +1,20 @@
-        if (__context_tag == to_underlying(Fields::k{{asUpperCamelCase label}}))
-        {
-            err = DataModel::Decode(reader, {{asLowerCamelCase label}});
-        }
-        else
-        {{#last}}
-        {
-        }
-
-        ReturnErrorOnFailure(err);
-        {{/last}}
-        {{/zcl_command_arguments}}
-    }
-}
-} // namespace {{asUpperCamelCase name}}.
-{{/zcl_commands}}
-} // namespace Commands
-
-namespace Attributes {
-CHIP_ERROR TypeInfo::DecodableType::Decode(TLV::TLVReader &reader, const ConcreteAttributePath &path) {
-    switch (path.mAttributeId)
-    {
-{{#zcl_attributes_server}}
-    case Attributes::{{asUpperCamelCase label}}::TypeInfo::GetAttributeId():
-        return DataModel::Decode(reader, {{asLowerCamelCase label}});
-{{/zcl_attributes_server}}
-    default:
-        return CHIP_NO_ERROR;
-    }
-}
-} // namespace Attributes
-
-namespace Events {
-{{#zcl_events}}
-namespace {{asUpperCamelCase name}} {
-CHIP_ERROR Type::Encode(TLV::TLVWriter & aWriter, TLV::Tag aTag) const{
-    TLV::TLVType outer;
-    ReturnErrorOnFailure(aWriter.StartContainer(aTag, TLV::kTLVType_Structure, outer));
-    {{#zcl_event_fields}}
-    {{#if_is_fabric_scoped_struct type}}
-    ReturnErrorOnFailure(DataModel::EncodeForRead(aWriter, TLV::ContextTag(Fields::k{{asUpperCamelCase name}}), GetFabricIndex(), {{asLowerCamelCase name}}));
-    {{else}}
-    ReturnErrorOnFailure(DataModel::Encode(aWriter, TLV::ContextTag(Fields::k{{asUpperCamelCase name}}), {{asLowerCamelCase name}}));
-    {{/if_is_fabric_scoped_struct}}
-    {{/zcl_event_fields}}
-    return aWriter.EndContainer(outer);
-}
-
-CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
-    detail::StructDecodeIterator __iterator(reader);
-    while (true) {
-        auto __element = __iterator.Next();
-        if (std::holds_alternative<CHIP_ERROR>(__element)) {
-           return std::get<CHIP_ERROR>(__element);
-        }
-
-        {{#zcl_event_fields}}
-        {{#first}}
-        CHIP_ERROR err = CHIP_NO_ERROR;
-        const uint8_t __context_tag = std::get<uint8_t>(__element);
-
-        {{/first~}}
-        {{! NOTE: using if/else instead of switch because it seems to generate smaller code. ~}}
-        if (__context_tag == to_underlying(Fields::k{{asUpperCamelCase name}}))
-        {
-            err = DataModel::Decode(reader, {{asLowerCamelCase name}});
-        }
-        else
-        {{#last}}
-        {
-        }
-
-        ReturnErrorOnFailure(err);
-        {{/last}}
-        {{/zcl_event_fields}}
-    }
-}
-} // namespace {{asUpperCamelCase name}}.
-{{/zcl_events}}
-} // namespace Events
-
-} // namespace {{asUpperCamelCase name}}
+#include <clusters/{{asUpperCamelCase name}}/Attributes.cpp>
+#include <clusters/{{asUpperCamelCase name}}/Commands.cpp>
+#include <clusters/{{asUpperCamelCase name}}/Events.cpp>
+#include <clusters/{{asUpperCamelCase name}}/Structs.cpp>
```

Comment:
I think our naming convention is to name "cpp meant to be included" as "ipp" or something like that?

Reply:
Updated to use ipp/